grype/match/ignore.go

```diff
@@ -241,14 +256,30 @@ func ifPackageLocationApplies(location string) ignoreCondition {
 }
 
 func ifUpstreamPackageNameApplies(name string) ignoreCondition {
-	pattern, err := packageNameRegex(name)
-	if err != nil {
-		log.WithFields("name", name, "error", err).Debug("unable to parse name expression")
-		return func(Match) bool { return false }
+	var (
+		pattern *regexp.Regexp
+		err     error
+	)
+	if strings.HasPrefix(name, "linux") {
```

Comment:
how come we dont care about -headers- here?

Reply:
https://github.com/anchore/grype/blob/main/cmd/grype/cli/commands/root.go#L106C1-L106C161

```
	{Package: match.IgnoreRulePackage{Name: "linux(-.*)?-headers-.*", [ Type: string(syftPkg.DebPkg)}, MatchType: match.ExactIndirectMatch},
```

Upstream cares about this part:
```
UpstreamName: "linux.*",
```